### PR TITLE
Threshold for permessage-deflate

### PR DESCRIPTION
**Pull request by nkzawa**
This PR adds the `threshold` option to perMessageDeflate for filtering data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -214,6 +214,7 @@ to a single process.
         (`true`)
       - `perMessageDeflate` (`Object|Boolean`): parameters of the WebSocket permessage-deflate extension
         (see [ws module](https://github.com/einaros/ws) api docs). Set to `false` to disable. (`true`)
+        - `threshold` (`Number`): data is compressed only if the byte size is above this value (`1024`)
```
Comment (nkzawa) [inline README.md]: We might be able to find a better default, but it's the same with http compression for now.

File: lib/socket.js [hunk 2]
```diff
@@ -216,7 +216,7 @@ Socket.prototype.maybeUpgrade = function (transport) {
   function check(){
     if ('polling' == self.transport.name && self.transport.writable) {
       debug('writing a noop packet to polling for fast upgrade');
-      self.transport.send([{ type: 'noop', options: { compress: false } }]);
+      self.transport.send([{ type: 'noop' }]);
     }
   }
```
Comment (rauchg) [inline lib/socket.js]: so much cleaner :) thank you

File: lib/transports/websocket.js [hunk 3]
```diff
@@ -87,8 +88,22 @@ WebSocket.prototype.send = function (packets) {
   packets.forEach(function(packet) {
     parser.encodePacket(packet, self.supportsBinary, function(data) {
       debug('writing "%s"', data);
+
+      // always creates a new object since ws modifies it
```
Comment (rauchg) [inline lib/transports/websocket.js]: is this a bug? why would `ws` mutate parameters?

Comment (nkzawa) [inline lib/transports/websocket.js]: It's not a bug of `ws`. Obviously, it's not good, but `ws` just works like that. 

https://github.com/nkzawa/ws/blob/master/lib/WebSocket.js#L230

Comment (nkzawa) [inline lib/transports/websocket.js]: Or we probably should call this a bug.

**Pull request merged**